packages/engine-core/src/data-proxy/errors/NetworkError.ts

```diff
@@ -0,0 +1,14 @@
+import type { DataProxyErrorInfo } from './DataProxyError'
+import { DataProxyError } from './DataProxyError'
+import { setRetryable } from './utils/setRetryable'
+
+export interface NetworkErrorInfo extends DataProxyErrorInfo {}
+
+export class NetworkError extends DataProxyError {
+  public name = 'NetworkError'
+  public code = 'P5010'
+
+  constructor(info: NetworkErrorInfo) {
+    super('Cannot fetch data from service', setRetryable(info, true))
```

Comment:
This error is retryable, so hopefully users will never notice unless there is downtime.